src/Network/Wai/SAML2/EntityDescriptor.hs

```diff
@@ -0,0 +1,62 @@
+{-# LANGUAGE LambdaCase #-}
+module Network.Wai.SAML2.EntityDescriptor
+    ( IDPSSODescriptor(..)
+    , Binding(..)
+    ) where
+
+import Data.Text (Text)
+import qualified Data.Map as Map
+import qualified Data.Text as T
+import Network.Wai.SAML2.XML
+import Text.XML.Cursor
+
+-- | Datatype describing metadata of an identity provider.
+data IDPSSODescriptor
+    = IDPSSODescriptor {
+        entityID :: Text
+        -- ^ IdP Entity ID. 'Network.Wai.SAML2.Config.saml2ExpectedIssuer' should be compared against this identifier
+    ,   x509Certificate :: Text
```

Comment:
In our usecase, we are storing the original x.509 certificates in the database. It should probably contain text representation, regardless of whether to parse it